pkg/util/error.go

```diff
@@ -43,23 +44,18 @@ func IsNotFound(err error) bool {
 	return false
 }
 
-func IsConnectionRefusedOrUnauthorizedOrAskingForCredentials(err error) bool {
+func IsConnectionRefusedOrUnauthorized(err error) bool {
 	if err == nil {
 		return false
 	}
 
-	msg := strings.ToLower(err.Error())
-	if strings.Contains(msg, "connection refused") || strings.Contains(msg, "unauthorized") || strings.Contains(msg, "the server has asked for the client to provide credentials") {
-		return true
-	}
-
-	return false
+	return errors.Is(err, syscall.ECONNREFUSED) || apierrors.IsUnauthorized(err)
 }
 
 func IsUnauthorized(err error) bool {
```

Comment:
Do we need this wrapper function? is `apierrors.IsUnauthorized(nil)` not return `false`?

Reply:
> Do we need this wrapper function?

Not necessarily I think. It could be considered a left over from our previous check.

>  is apierrors.IsUnauthorized(nil) not return false?

I think it does return `false` here.

If you prever that, I can remove this wrapper function and directly call `apierrors.IsUnauthorized`. The only place we use it so far is [here](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/controller/kyma_controller.go#L135)

Reply:
exactly, basically, you can write a small unit test to verify it, something like:
```
func Test_unauth(t *testing.T) {
	assert.Equal(t, false, apierrors.IsUnauthorized(nil))
}
```

so let's remove this function, and use `apierrors.IsUnauthorized` directly.

Reply:
https://github.com/kyma-project/lifecycle-manager/pull/1313

> exactly, basically, you can write a small unit test to verify it, something like

did that and works as expected. I also assume that you just meant for trying this out, not actually putting a unit test like this into the project.